OneCommunityGlobal/HGNRest
# Fix noInfringementStreak badges

# Description
There is a badge that awards users who go to certain lengths of months without acquiring or using a blue square. These badges were not being awarded despite the data and parameters being met. Testing this bug requires some patience as it takes a couple of minutes for badges to be updated on a user profile. You may need to switch between the dashboard and profile page as well as reload the app to see the new badge. You can also check if the badge is added by keeping an eye on the terminal log for the backend to see if “Added Badge” or “Replaced Badge” is logged. 

Fixes # (Badge Component #9 “no blue square for x months”)

## Mainly changes explained:

The bug originated from a misspelling with an incorrect camel case of a property of the user object.
I changed the variable from oldinfringements to oldInfringements and the function is working properly. 


## How to test:
check into current branch
do `npm install` and `...` to run this PR locally
log in as an admin user or any other user that can adjust start dates and blue squares
go to dashboard→ Profile→ Volunteer times→ Start Date
Remove all blue squares from the user
Change start date to simulate 3, 6, 12, 24, 36, and 48 months spans of time
Wait for the userProfile cronjob to run and award badges. Check for logged badge addition or replacement
Wait 5-10 min for the badge to update on the user. (May need to reload or switch pages)

## Screenshots or videos of changes:

## Note:
I noticed that even though the badges are added when the cronjob is run, it is not updated on the app until the second run or the cronjob sometimes even later.
